comp/dogstatsd/server/parse.go

```diff
@@ -81,7 +79,7 @@ func newParser(cfg model.Reader, float64List *float64ListPool, workerNum int, wm
 		readTimestamps:   readTimestamps,
 		float64List:      float64List,
 		dsdOriginEnabled: cfg.GetBool("dogstatsd_origin_detection_client"),
-		provider:         provider.GetProvider(wmeta),
+		provider:         provider.GetMetaCollector(),
```

Comment:
This seems unnecessarily fragile. Before this change the provider was made sure to be initialized whenever it was used, now this depends on the code scattered around `cmd` for this. (`GetProvider` starts a background process that feeds data into the meta collector, but `GetMetaCollector` does not).  I get the need to decouple the tagger component from the workloadmeta, but is there really any need to decouple the dogstatsd server?